bugsnag-android-core/src/main/java/com/bugsnag/android/BugsnagPluginInterface.kt

```diff
@@ -9,6 +9,15 @@ interface BugsnagPlugin {
 object BugsnagPluginInterface {
 
     private var plugins = mutableMapOf<Class<*>, BugsnagPlugin>()
+    private var registeredPluginClasses = mutableSetOf<Class<*>>()
+
+    fun registerPlugin(clz: Class<*>) {
```

Comment:
These methods should either have `protected` access visibility or `internal` with the `@JvmName` annotation, as this will avoid exposing them as part of our public API.

Reply:
Is this class public? Are things public by default?

Reply:
Yes, in Kotlin everything is public by default. Other than that, the modifiers are reasonably similar to Java but have some subtle differences: https://kotlinlang.org/docs/reference/visibility-modifiers.html

Reply:
I see. So when this function was removed in #603, it broke a public interface. 😞 I did not realize that at the time.

Reply:
I've made an update to protect `loadRegisteredPlugins` as it should only be called from within the same module. However, I've left `registerPlugin` as-is because calling it from a class which was not compiled as a part of the same source set means its not in the same module:

> The internal visibility modifier means that the member is visible within the same module. More specifically, a module is a set of Kotlin files compiled together:
> * an IntelliJ IDEA module;
> * a Maven project;
> * a Gradle source set (with the exception that the test source set can access the internal declarations of main);
> * a set of files compiled with one invocation of the <kotlinc> Ant task.

There doesn't seem to be a package access modifier or equivalent.

Reply:
Yep - this is the right approach as we need to expose `registerPlugin` to other modules. The only alternative I'm aware of would be reflection